diff --git a/wm.js b/wm.js
index daa261f..fc5323c 100644
--- a/wm.js
+++ b/wm.js
@@ -386,10 +386,19 @@ var WindowManager = GObject.registerClass(
                 target_rect.height *= size;
                 target_rect.height -= target_rect.height % monitor_scale;
 
-                if (this.right_or_bottom)
-                    target_rect.y += workarea.height - target_rect.height;
+                const padding = 10;
+                
+                target_rect.width /= 2;
+                target_rect.x += target_rect.width / 2;
+
+                if (this.right_or_bottom) {
+                    target_rect.y += workarea.height - target_rect.height - padding;
+                } else {
+                    target_rect.y += padding;
+                }
             }
 
+
             return target_rect;
         }
 
